src/Projection/Projection/Store/DoctrineStore.php

```diff
@@ -55,94 +59,111 @@ public function get(string $projectionId): Projection
         return $this->createProjection($result);
     }
 
-    public function all(): ProjectionCollection
+    /** @return list<Projection> */
+    public function find(ProjectionCriteria|null $criteria = null): array
     {
-        $sql = $this->connection->createQueryBuilder()
+        $qb = $this->connection->createQueryBuilder()
             ->select('*')
-            ->from($this->projectionTable)
-            ->getSQL();
+            ->from($this->projectionTable);
+
+        if (!$this->connection->getDatabasePlatform() instanceof SQLitePlatform) {
+            $qb->forUpdate();
```

Comment:
First, i thought about an interface for providing locking and using a "dbal" implementation here, but maybe this is too much? 

Second, checking on the platform here is maybe a bit brittle, no? DBAL could easily add a new platform which also does not support this feature and then we would not catch this with our check here unless updated.

Reply:
> First, i thought about an interface for providing locking and using a "dbal" implementation here, but maybe this is too much?

I thought about it too, but there were too many if statements at the time. But now I could do it because I abstracted the part well in the projectionist.

> Second, checking on the platform here is maybe a bit brittle, no? DBAL could easily add a new platform which also does not support this feature and then we would not catch this with our check here unless updated.

It doesn't really matter whether DBAL adds a new platform or whether SQLite implements the feature in the future. We listed the supported DB versions in the Readme and expressly said that we only support them. These are also tested in the CI.

I can't solve it any other way. There is no "supportForUpdate" method.